src/main/java/pl/sda/weather/controllers/LocationController.java

```diff
@@ -2,57 +2,61 @@
 
 
 import pl.sda.weather.model.LocationModel;
+import pl.sda.weather.model.Weather;
 import pl.sda.weather.service.ILocationService;
 import pl.sda.weather.service.impl.LocationServiceImpl;
-import pl.sda.weather.validation.LocationValidation;
+import pl.sda.weather.service.impl.ReadWeathersImpl;
+import pl.sda.weather.validation.LocationValidator;
 
 import java.util.List;
+import java.util.Map;
 import java.util.Scanner;
 import java.util.UUID;
 
 public class LocationController {
-    ILocationService locationService = new LocationServiceImpl();
+
+    private static final ILocationService locationService = new LocationServiceImpl();
 
     private static final Scanner scanner = new Scanner(System.in);
 
-    private static LocationValidation validation = new LocationValidation();
+    private static LocationValidator validation = new LocationValidator();
+
+    ReadWeathersImpl readWeathers = new ReadWeathersImpl();
 
 
     public void addLocation() {
 
         String idUUID = String.valueOf(UUID.randomUUID());
 
-       String longitudeAndLatitude = "lok";
-//        String longitudeS;
-//        String longitudeN;
-//        String latitudeW;
-//        String latitudeE;
-//       do {
-//           System.out.println("Write longitude S :");
-//           longitudeS = scanner.nextLine();
-//
-//           System.out.println("Write longitude  N :");
-//           longitudeN = scanner.nextLine();
-//
-//           System.out.println("Write latitude W :");
-//           latitudeW = scanner.nextLine();
-//
-//           System.out.println("Write latitude E :");
-//           latitudeE = scanner.nextLine();
-//
-//           longitudeAndLatitude = "(latitude: " + longitudeS + " -> S; " + longitudeN + " -> N; longitude: " + latitudeW + " -> W; " + latitudeE + " -> E)";
-//
-//
-//       }while (latitudeE.equals("") || latitudeW.equals("") || longitudeN.equals("") || longitudeS.equals(""));
+        String longitude;
+        String latitude;
+
+        do {
+            System.out.println("Write city latitude in format (000.00) : ");
```

Comment:
same as before -> classes related with View, should be in View class (not in Controller).